plugins/auth-backend-module-openshift-provider/src/resolvers.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  createSignInResolverFactory,
+  OAuthAuthenticatorResult,
+  PassportProfile,
+  SignInInfo,
+} from '@backstage/plugin-auth-node';
+
+import {
+  DEFAULT_NAMESPACE,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+
+export namespace openshiftSignInResolvers {
+  export const displayNameMatchingUserEntityName = createSignInResolverFactory({
+    create() {
+      return async (
+        info: SignInInfo<OAuthAuthenticatorResult<PassportProfile>>,
+        ctx,
+      ) => {
+        const { displayName } = info.profile;
+
+        if (!displayName) {
+          throw new Error(
+            `OpenShift user profile does not contain a displayName`,
+          );
+        }
+
+        const userRef = stringifyEntityRef({
+          kind: 'User',
+          name: displayName,
+          namespace: DEFAULT_NAMESPACE,
+        });
+
+        return await ctx.signInWithCatalogUser({ entityRef: userRef });
```

Comment:
Thanks for updating this! One more thing, since [this PR](https://github.com/backstage/backstage/pull/28967) to add the `dangerouslyAllowSignInWithoutUserInCatalog` config got merged, do you mind rebasing and supporting this config with this resolver as well like [this](https://github.com/JessicaJHee/backstage/blob/ab53e6fed475e203791f1654e51d43e1cec52c1f/plugins/auth-backend-module-github-provider/src/resolvers.ts#L52-L62)?

Reply:
Yes I can do that 👍

Reply:
I implemented `dangerouslyAllowSignInWithoutUserInCatalog` and may not suggest it in production. It is pretty neat to use for local testing. 🎉 

Hope everything is okay and looking forward to your review @JessicaJHee.